Makefile

```diff
@@ -57,3 +58,7 @@ clean :
 	@rm -f ${BIN}
 	@echo "${BIN} cleaned"
 
+docker :
+
+	${DOCKER} build -t localhost/tty-clock .
```

Comment:
"localhost"? that looks odd anyways...

how is this container going to be published and maintained in the long term? maybe we need an action here?

Reply:
localhost is just a tag that is reserved and cannot be pushed. I hadn't thought about pushing it anywhere, but you definitely could.

Reply:
I made the image configurable now via the DOCKER_IMAGE variable, can build an alternate tag this way:

```
DOCKER_IMAGE=xorg62/tty-clock make docker
```

Reply:
Because docker can build directly from git, I almost never use third party images anymore. I just always build it from source. You can do the same thing with `docker-compose` if you replace all `image` references with `build: context: GIT_URL` instead ([live example](https://github.com/EnigmaCurry/d.rymcg.tech/blob/bc62b7dc218c80170c316e1e32c0973c7e2e54f2/maubot/docker-compose.yaml#L27-L30) If I were deploying to a bunch of servers, I might want to use an image from a registry, but it takes only two seconds to build this image, so I just do it on the fly (with one command: `docker-compose up` knows to build it if hasn't been built already). I feel a lot more comfortable with the supply chain of the docker image this way too, rather than relying on some one elses image.

Reply:
One more example, this is with tty-clock on my fork:

https://github.com/EnigmaCurry/d.rymcg.tech/blob/master/_terminal/tty-clock/Makefile

All you need to run the clock is this Makefile, notice there is no other files in the directory at all :) Docker builds everything from github directly and stores the image locally. Deterministic builds are so cool.